pages/workspaces/[workspaceId]/activity.tsx

```diff
@@ -163,32 +158,36 @@ const WorkspaceActivityPage = ({
                     setQueryParams({ page: "1" });
                   }}
                 />
-                {isMobile ? null : <DayRangePicker />}
+                <ClientOnly>{isMobile ? <DayRangePicker /> : null}</ClientOnly>
                 <LimitPicker />
               </div>
             </div>
             <ClientOnly>
               <WorkspacePullRequestTable isLoading={isLoading} data={pullRequests} meta={meta} />
             </ClientOnly>
           </div>
-          <InsightUpgradeModal
-            workspaceId={workspace.id}
-            variant="contributors"
-            isOpen={isInsightUpgradeModalOpen}
-            onClose={() => setIsInsightUpgradeModalOpen(false)}
-            overLimit={10}
-          />
         </div>
       </WorkspaceLayout>
-      <StarSearchEmbed
-        userId={userId}
-        bearerToken={bearerToken}
-        suggestions={WORKSPACE_STARSEARCH_SUGGESTIONS}
-        isMobile={isMobile}
-        // TODO: implement once we have shared chats in workspaces
-        sharedChatId={null}
-        tagline="Ask anything about your workspace"
+      <ClientOnly>
```

Comment:
This could be a dynamic import that loads client-side only but I noticed if it takes a while for data to load in grids client-side, it delays the loading of this, so I decided to keep it part of the server-side render.